sdks/java/extensions/sketching/src/main/java/org/apache/beam/sdk/extensions/sketching/TDigestQuantiles.java

```diff
@@ -0,0 +1,345 @@
+    abstract static class Builder<K> {
+      abstract Builder<K> setCompression(double cf);
+
+      abstract PerKeyDigest<K> build();
+    }
+
+    /**
+     * Returns a new {@link PTransform} with a new compression factor {@code cf}.
+     *
+     * <p>This value bounds the maximum number of points resumed in each centroid
+     * and the total number of centroids in the digest.
+     *
+     * @param cf the bound value for centroid and digest sizes.
+     */
+    public PerKeyDigest<K> withCompression(double cf) {
+      return toBuilder().setCompression(cf).build();
+    }
+
+    @Override
+    public PCollection<KV<K, MergingDigest>> expand(PCollection<KV<K, Double>> input) {
+      return input.apply(
+              "Compute T-Digest Structure",
+              Combine.perKey(TDigestQuantilesFn.create(this.compression())));
+    }
+  }
+
+  /** Implements the {@link Combine.CombineFn} of {@link TDigestQuantiles} transforms. */
+  public static class TDigestQuantilesFn
+      extends Combine.CombineFn<Double, MergingDigest, MergingDigest> {
+
+    private final double compression;
+
+    private TDigestQuantilesFn(double compression) {
+      this.compression = compression;
+    }
+
+    /**
+     * Returns {@link TDigestQuantilesFn} combiner with the given compression factor.
+     *
+     * <p>This value bounds the maximum number of points resumed in each centroid
+     * and the total number of centroids in the digest.
+     *
+     * @param compression the bound value for centroid and digest sizes.
+     */
+    public static TDigestQuantilesFn create(double compression) {
+        if (compression > 0) {
+            return new TDigestQuantilesFn(compression);
+        }
+        throw new IllegalArgumentException("Compression factor should be greater than 0.");
+    }
+
+    @Override public MergingDigest createAccumulator() {
+      return new MergingDigest(compression);
+    }
+
+    @Override public MergingDigest addInput(MergingDigest accum, Double value) {
+      accum.add(value);
+      return accum;
+    }
+
+    /** Output the whole structure so it can be queried, reused or stored easily. */
+    @Override public MergingDigest extractOutput(MergingDigest accum) {
+      return accum;
+    }
+
+    @Override public MergingDigest mergeAccumulators(
+        Iterable<MergingDigest> accumulators) {
+      Iterator<MergingDigest> it = accumulators.iterator();
+      MergingDigest merged = it.next();
+      while (it.hasNext()) {
+        merged.add(it.next());
+      }
+      return merged;
+    }
+
+    @Override public Coder<MergingDigest> getAccumulatorCoder(CoderRegistry registry,
+        Coder inputCoder) {
+      return new MergingDigestCoder();
+    }
+
+    @Override public Coder<MergingDigest> getDefaultOutputCoder(CoderRegistry registry,
+        Coder inputCoder) {
+      return new MergingDigestCoder();
+    }
+
+    @Override public MergingDigest defaultValue() {
+      return new MergingDigest(10);
```

Comment:
Why 10?

Reply:
Oh I forgot this, it was from the first versions but I didn't set a default value for other sketches so I can even delete this one what do you think ? Otherwise I will put the same default value as on the builders.

Reply:
You can delete this one. Overriding it is necessary in a combine-with-context, otherwise it's equivalent to extractOutput(createAccumulator()).

Reply:
Ok, I also added populateDisplayData() method as I did in other sketches.